File: src/main/java/fr/fabienhebuterne/marketplace/storage/Repository.java
package fr.fabienhebuterne.marketplace.storage;

import java.util.List;

public interface Repository<T> {
    public List<T> findAll();
    public T find(int id);
    public T create(T entity);
    public T update(int id, T entity);
    public boolean delete(int id);
}

File: src/main/kotlin/fr/fabienhebuterne/marketplace/storage/Repository.kt
package fr.fabienhebuterne.marketplace.storage

import fr.fabienhebuterne.marketplace.domain.base.Filter
import org.jetbrains.exposed.sql.ResultRow
import org.jetbrains.exposed.sql.statements.UpdateBuilder
import java.util.*

interface Repository<T> {
    fun fromRow(row: ResultRow): T
    fun fromEntity(insertTo: UpdateBuilder<Number>, entity: T): UpdateBuilder<Number>
    fun findAll(from: Int?, to: Int?, searchKeyword: String?, filter: Filter): List<T>
    fun find(id: String): T?
    fun create(entity: T): T
    fun update(entity: T): T
    fun delete(id: UUID)
    fun countAll(searchKeyword: String?): Int
}